### PR TITLE
[RFR] Bootstrap and first test

### PR DESCRIPTION
**Pull request by benjaminpayoux**
# TODO
- [X] Add readme and makefile
- [X] Install the project in docker
- [X] Run a test with make test
- [X] Run a linter with make lint

# Result
last

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.12.6
```
Comment (mchaffotte) [inline Dockerfile]: Why do not use the last version of go ?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	docker-compose build
+
+start: ## Start the server
+	docker-compose up
+
+stop: ## Stop the server
+	docker-compose down
+
+test: ## Tests the API
+	docker-compose run --rm --no-deps api go test -v ./test/...
```
Comment (mchaffotte) [inline Makefile:16]: You can add a command to lint your go code

**Review by mchaffotte — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (benjaminpayoux) [reply Dockerfile]: Right. Updated :+1:

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# Download all the dependencies
+# https://stackoverflow.com/questions/28031603/what-do-three-dots-mean-in-go-command-line-invocations
+RUN go get -d -v github.com/gorilla/mux
+RUN go get -d -v github.com/patrickmn/go-cache
+RUN go get -d -v github.com/lithammer/shortuuid
+
+# Install the package
+RUN go install -v github.com/gorilla/mux
+RUN go install -v github.com/patrickmn/go-cache
+RUN go get -d -v github.com/lithammer/shortuuid
```
Comment (HALLERPierre) [inline Dockerfile]: You get `shortuuid` 2 times but you don't install it

Comment (JulienMattiussi) [inline Dockerfile]: I think it's a too fast copy/past of previous line

**Review by HALLERPierre — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (benjaminpayoux) [reply Dockerfile]: Removed in the last commit

Comment (benjaminpayoux) [reply Dockerfile]: Removed in last commit

File: test/battleship/game/game_test.go [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package game
```
Comment (mchaffotte) [inline test/battleship/game/game_test.go]: This file does not test main.go

Comment (benjaminpayoux) [reply test/battleship/game/game_test.go]: I just bootstrap the run of an empty test. Real tests will be added in next commits.

Comment (mchaffotte) [reply test/battleship/game/game_test.go]: Ok so remove it or test something existing. Do no add useless files

**Review by mchaffotte — APPROVED** [review]

**Pull request merged**